Adds recipe for github-topics

agzam opened the pull request:
### Brief summary of what the package does

github-topics is a small and very simple package that allows for quickly searching for Pull-Requests matching a string.

The name is such, because I'm hoping to add more functionality at some point

### Direct link to the package repository

https://github.com/agzam/github-topics

### Your association with the package

Author

### Relevant communications with the upstream package maintainer

**None needed**

### Checklist

<!-- Please confirm by replacing `[]` with `[x]`: -->

- [x] The package is released under a [GPL-Compatible Free Software License](https://www.gnu.org/licenses/license-list.en.html#GPLCompatibleLicenses)
- [x] I've read [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [x] I've used the latest version of [package-lint](https://github.com/purcell/package-lint) to check for packaging issues, and addressed its feedback
- [x] My elisp byte-compiles cleanly
- [x] I've used `M-x checkdoc` to check the package's documentation strings
- [x] I've built and installed the package using the instructions in [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)

<!-- After submitting, please fix any problems the CI reports. -->

riscy commented:
Thanks for this.  This looks handy, but I feel like your summary/`;;; Commentary` is not describing a key aspect of this, which is that it fetches the PR data (matching a query) and puts everything in an org buffer.  Also the extra "Description" line isn't necessary, e.g.
```elisp
;;  Description
;;
;;  For finding Pull-Requests on GitHub using 'gh' command-line tool.
```
should be just:
```
;;  For finding Pull-Requests on GitHub using 'gh' command-line tool.
;; The results are displayed in an org-mode buffer, etc etc
;; Once this package is loaded, run the command xyz to etc etc
```

Beyond that, quick first pass -

⸺ `github-topics.el` with [melpazoid](https://github.com/riscy/melpazoid):
```
- github-topics.el#L68: Use `when ...` instead of `unless (null ...)`
```

⸺ Package and license:
- Please specify `:fetcher` before `:repo` in your recipe
- github-topics.el needs *formal* license boilerplate and/or an [SPDX-License-Identifier](https://spdx.dev/ids/)

agzam commented:
Thank you @riscy for such a thorough review, appreciate it. I have addressed your comments:

- in the header I added description in the Commentary section and completely removed the Description part. 

- fixed the `unless null` thingy

- moved `:fetcher` to the front of the recipe

- added the license identifier

Let me know if you spot some other things.

riscy commented:
Thank you!  I'll merge this now.
